### PR TITLE
Change affinity parameter to metric

### PR DESCRIPTION
**Pull request by gc153671**
AgglomerativeClustering parameter was renamed to metric and [fully deprecated in sklearn 1.4](https://scikit-learn.org/1.3/modules/generated/sklearn.cluster.AgglomerativeClustering.html#sklearn-cluster-agglomerativeclustering)

### CODE REVIEW

**Review by jc-healy — APPROVED** [review]
Looks good.  Thanks for the PR
